src/Orleans/CodeGeneration/TypeUtils.cs

```diff
@@ -561,7 +561,7 @@ public static IEnumerable<TypeInfo> GetDefinedTypes(Assembly assembly, Logger lo
             }
             catch (Exception exception)
             {
-                if (logger.IsWarning)
+                if (logger!= null && logger.IsWarning)
                 {
                     var message =
                         string.Format(
```

Comment:
Can we move to use string interpolation when adding new code? It would be less to cleanup at a later stage.